linera-execution/src/evm/database.rs

```diff
@@ -209,17 +211,54 @@ where
 {
     type Error = ExecutionError;
 
+    /// The `basic_ref` is the function for reading the state of the application.
     fn basic_ref(&self, address: Address) -> Result<Option<AccountInfo>, ExecutionError> {
+        if address == FAUCET_ADDRESS {
```

Comment:
What's this for? Why is there a "faucet" address with constant balance?

Reply:
The EVM has a representation of the state, which makes it think that it controls everything. But that is not so, because we have other chains. But when we do a transfer, it has to go somewhere. And that somewhere is the faucet.

We see analogous constructions in ERC20 smart contract in order to handle burning/minting, we transfer to a special address 0 which is not used by anyone.

Reply:
Yes, reading the rest of the code that's what I suspected. But it would be good to write that explicitly in a comment, maybe where the constant `FAUCET_ADDRESS` is defined?

Reply:
I don't see the analogy between burning and hardcoding a facuet address. "burning" tokens is a cryptographic construct – it's about sending tokens to an address that cannot have a corresponding private key. It's also "economical" construct: such tokens don't decrease the total supply but decrease circulating supply (cannot be moved/used) so they increase the value of the rest of the tokens in the circulation.